pandas/core/groupby/ops.py

```diff
@@ -657,7 +657,13 @@ def _aggregate_series_pure_python(self, obj: Series, func):
             res = func(group)
             if result is None:
                 if isinstance(res, (Series, Index, np.ndarray)):
-                    raise ValueError("Function does not reduce")
+                    if len(res) == 1:
+                        # e.g. test_agg_lambda_with_timezone lambda e: e.head(1)
+                        # FIXME: are we potentially losing import res.index info?
+                        res = getattr(res, "_values", res)
```

Comment:
can you just do
```res = np.array(res)[0]```? or
```res = res.item()``` (though I think we have deprecated ``.item()``, but are bringing it back.

Reply:
`np.array` won't work because itll lose the timezone.  im thinking `next(iter(res))` with a comment saying its often equivalent to `res[0]`

Reply:
I think `.item()` would be good here as well

Reply:
`.item()` will be good if/when we un-deprecate it

Reply:
is the current `next(iter(res))` acceptable for the time being?  `extract_array(res)[0]` would also work.

I'm eager to see this go in because I think we can get rid of a bunch more _try_cast calls

Reply:
yep just mark it with the issue (or fix me) so it’s clear

Reply:
commented